accounts/abi/bind/precompiletemplate.go

```diff
@@ -0,0 +1,311 @@
+
+	{{.Contract.Type}}Precompile StatefulPrecompiledContract = create{{.Contract.Type}}Precompile({{.Contract.Type}}Address)
+	{{- range .Contract.Funcs}}
+	{{decapitalise .Normalized.Name}}Signature = CalculateFunctionSelector("{{.Original.Sig}}")
+	{{- if not .Original.IsConstant}}
+
+	ErrCannot{{.Normalized.Name}} = errors.New("non-enabled cannot {{.Original.Name}}")
+	{{- end}}
+	{{- end}}
+
+	{{.Contract.Type}}ABI abi.ABI // will be filled by init func
+)
+
+// {{.Contract.Type}}Config implements the StatefulPrecompileConfig
+// interface while adding in the {{.Contract.Type}} specific precompile address.
+type {{.Contract.Type}}Config struct {
+	{{- if .Contract.AllowList}}
+	AllowListConfig
+	{{- end}}
+	UpgradeableConfig
+}
+
+{{$structs := .Structs}}
+{{range $structs}}
+	// {{.Name}} is an auto generated low-level Go binding around an user-defined struct.
+	type {{.Name}} struct {
+	{{range $field := .Fields}}
+	{{$field.Name}} {{$field.Type}}{{end}}
+	}
+{{- end}}
+
+{{- range .Contract.Funcs}}
+{{ if len .Normalized.Inputs | lt 1}}
+type {{capitalise .Normalized.Name}}Input struct{
+{{range .Normalized.Inputs}} {{capitalise .Name}} {{bindtype .Type $structs}}; {{end}}
+}
+{{- end}}
+{{ if len .Normalized.Outputs | lt 1}}
+type {{capitalise .Normalized.Name}}Output struct{
+{{range .Normalized.Outputs}} {{capitalise .Name}} {{bindtype .Type $structs}}; {{end}}
+}
+{{- end}}
+{{- end}}
+
+func init() {
+	parsed, err := abi.JSON(strings.NewReader({{.Contract.Type}}RawABI))
+	if err != nil {
+		panic(err)
+	}
+	{{.Contract.Type}}ABI = parsed
+}
+
+// New{{.Contract.Type}}Config returns a config for a network upgrade at [blockTimestamp] that enables
+// {{.Contract.Type}} {{if .Contract.AllowList}} with the given [admins] as members of the allowlist {{end}}.
+func New{{.Contract.Type}}Config(blockTimestamp *big.Int{{if .Contract.AllowList}}, admins []common.Address{{end}}) *{{.Contract.Type}}Config {
+	return &{{.Contract.Type}}Config{
+		{{if .Contract.AllowList}}AllowListConfig:   AllowListConfig{AllowListAdmins: admins},{{end}}
+		UpgradeableConfig: UpgradeableConfig{BlockTimestamp: blockTimestamp},
+	}
+}
+
+// NewDisable{{.Contract.Type}}Config returns config for a network upgrade at [blockTimestamp]
+// that disables {{.Contract.Type}}.
+func NewDisable{{.Contract.Type}}Config(blockTimestamp *big.Int) *{{.Contract.Type}}Config {
+	return &{{.Contract.Type}}Config{
+		UpgradeableConfig: UpgradeableConfig{
+			BlockTimestamp: blockTimestamp,
+			Disable:        true,
+		},
+	}
+}
+
+// Equal returns true if [s] is a [*{{.Contract.Type}}Config] and it has been configured identical to [c].
+func (c *{{.Contract.Type}}Config) Equal(s StatefulPrecompileConfig) bool {
+	// typecast before comparison
+	other, ok := (s).(*{{.Contract.Type}}Config)
+	if !ok {
+		return false
+	}
+	return c.UpgradeableConfig.Equal(&other.UpgradeableConfig) {{if .Contract.AllowList}} && c.AllowListConfig.Equal(&other.AllowListConfig) {{end}}
```

Comment:
Should we add a TODO here to add whatever additional checks are necessary to ensure that the configs are actually equal?

And clarify that if the timestamp / allow list is the only thing in the config then no additional changes are necessary.